### DISCUSSION TITLE
Build headless server for Ubuntu

### DISCUSSION THREAD

**Discussion by mcfnord** [jamulussoftware/jamulus#3104] [Technical Support & Help] [answered]
I want to [add this code](https://github.com/dtinth/jamulus/commit/d8469480b0f9408b5b46ad3f668ae1b95fc73955). 
 
Using equivalent code for a client, I make one server at a time reveal the song title & artist, or whether there's a Jitsi video link. 

![image](https://github.com/jamulussoftware/jamulus/assets/1750243/9437ae4e-c0c5-48b7-abc8-31f186b1cfc5)
![image](https://github.com/jamulussoftware/jamulus/assets/1750243/eb6dd04c-1a57-4973-9f8e-2c950468b393)

I only need to run the server headlessly, and only on Ubuntu (ARM and AMD). And I can build these bits on Ubuntu or Windows, whichever is easier. And I bet it's easier still to just build the executable. I could install the normal deb file and overwrite with the modified executable.

That's the tiny scope of what I actually need. What is the simplest approach that could possibly work for me?

**Comment by pljones** [comment]
If you want multi-platform builds, in my view getting Github to do it via the build process is easiest.

Fork, apply any patches you want, update to a release version, merge everything to your fork's master and push to Github.  Then, grab the built artifact for the platforms you're after and either use the generated installers or use the normal platform tools to pull out the Jamulus executable from the installer.  I know that's possible for the .deb files.  Not so sure on the Windows side.

**Reply by pljones** [reply]
Actually, you don't need to push a release version now I think about it - just raise a PR against your fork's master from a branch should do the build.

**Reply by ann0see** [reply]
Even just pushing a Tag beginning with "r" should've enough

**Reply by mcfnord** [reply]
I didn't know my fork could use its origin's build system! This sounds kinda easy!

**Reply by mcfnord** [reply]
I made a PR for my fork's master. It even had a label called release-me. If a label is same as a tag.

It said continuous integration wasn't set up.

![image](https://github.com/jamulussoftware/jamulus/assets/1750243/08a56e3b-abd7-4868-bf06-d44893fce690)

Do I need to set up continuous integration? Is that a GitHub Action? How do you set up a PR on your fork to trigger a build?

**Reply by ann0see** [reply]
Hmm. Probably you need to enable GitHub actions in the settings of your repo.

**Reply by mcfnord** [reply, marked as answer]
Ok, I enabled Actions. But everything's empty. 

I found this:

https://github.com/jamulussoftware/jamulus/blob/main/.github/workflows/autobuild.yml

(There's also an autobuild folder that appears to cover differentt OS builds. Maybe that script triggers those scripts.)

Ok, that's not in my fork. But it looks like maybe it's how you auto-build. And does this work on your fork?

Maybe I could copy... but you're saying your fork gets these for free?

I seem to get the same Actions categories as you... so maybe I do get these actions. So I clicked Run Workflow on Auto-Build... and maybe it's working!!! NOW IT SAYS IT'S BUILDING 5 PLATFORMS??? THIS IS EXCITING! OMG IT SAYS LINUX .DEB IS DONE... BUT haven't created .deb files yet. EXCITEMENT!!!

Windows failed with c1xx: fatal error C1356: unable to find mspdbcore.dll when compiling moc_serverlist.cpp ? I only want Linux so ok!

A final step, Create files for .deb repo, was skipped. But maybe Build for Linux .deb amd64 completed... now I hunt for the build artifacts!!

"Artifact jamulus-headless_3.10.0dev-89c8a7b_ubuntu_amd64.deb has been successfully uploaded!" ... somewhere... 

OMG OMG OMG

![image](https://github.com/jamulussoftware/jamulus/assets/1750243/d465de71-d633-4dd7-b236-612eaecd181f)
